src/NuGet.Core/NuGet.Frameworks/CompatibilityProvider.cs

```diff
@@ -182,16 +182,39 @@ private bool IsCompatibleWithTarget(NuGetFramework target, NuGetFramework candid
 
         private static bool IsCompatibleWithTargetCore(NuGetFramework target, NuGetFramework candidate)
         {
-            bool result = NuGetFramework.FrameworkNameComparer.Equals(target, candidate)
-                    && IsVersionCompatible(target.Version, candidate.Version)
-                    && StringComparer.OrdinalIgnoreCase.Equals(target.Profile, candidate.Profile);
-
-            if (target.IsNet5Era && candidate.HasPlatform)
+            bool result = true;
+            bool isNet6Era = StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, target.Framework) && target.Version.Major >= 6;
+            if (isNet6Era && target.HasPlatform && !NuGetFramework.FrameworkNameComparer.Equals(target, candidate))
             {
-                result = result
-                    && StringComparer.OrdinalIgnoreCase.Equals(target.Platform, candidate.Platform)
-                    && IsVersionCompatible(target.PlatformVersion, candidate.PlatformVersion);
+                if (candidate.Framework.StartsWith("xamarin.", StringComparison.OrdinalIgnoreCase))
+                {
+                    string suffix = candidate.Framework.ToUpperInvariant().Replace("XAMARIN.", "");
```

Comment:
I'm not too worried, but this is a relative hot codepath during restore, so I'd consider if you can make it allocate less. 
For example, you know how long the string you are truncating is,

There are plenty of compare APIs that can do this with minimal allocations. 
https://docs.microsoft.com/en-us/dotnet/api/system.string.compare?view=net-5.0#System_String_Compare_System_String_System_Int32_System_String_System_Int32_System_Int32_

Reply:
This check has been removed